test/decl/var/properties.swift

```diff
@@ -1340,3 +1340,8 @@ class LazyPropInClass {
   lazy var foo: Int = { return 0 } // expected-error {{function produces expected type 'Int'; did you mean to call it with '()'?}}
   // expected-note@-1 {{Remove '=' to make 'foo' a computed property}}{{21-23=}}{{3-8=}}
 }
+
+// SR-15657
+enum SR15657 {
+  var foo: Int {} // expected-error{{computed property must have accessors specified}}
```

Comment:
What does this expected error mean? Is it suggesting that explicit `get { ... }` is required? If so, that's misleading since it's not actually required—there's just no suitable return statement.

Reply:
That is the pre-existing diagnostic there and from context my understanding that is suggesting this is an incomplete computed property and it has to be at least one accessor specified or a result computed expression. Maybe we could improve the wording here, do you have any suggestion?

Reply:
Well, the error doesn't mention anything about a return value. There's only one way I can interpret the word "must," which is that _there is no other way_ to make this code compile (besides deleting what's written) except to "specify accessors." This would have to mean that the user hasn't already specified an accessor already (which they have, implicitly) and/or that they must specify one explicitly (which they don't have to). So I think this is still a misleading diagnostic for an incomplete computed property.

In my view, the compiler should be giving the same diagnostic as for any function that's supposed to return `Int` but is empty:

> Missing return in an accessor expected to return 'Int'.

Reply:
Awesome, I like the suggestion! But was thinking also about using `return` wording as it is not always a `return` and can be a `yield` in `_read`|`_modify` accessors.
What about using `result` something like: `missing accessor or return in computed property expected to result in a value of type 'Int'`?

Reply:
Well, this accessor is a getter, so it _is_ expected to return.

Reply:
> I think you’re saying we should still diagnose this as though it’s a container for arbitrary accessors

Yeah, exactly

> By the way, it wouldn’t be correct to assume that the user wants a computed property if you don’t assume the user intends to write an implicit getter

I understand the point specially now that you pointed other possible accessors are underscored... I'll make the changes.